system/jlib/jthread.cpp

```diff
@@ -607,6 +607,7 @@ void CThreadedPersistent::threadmain()
             break;
         try
         {
+            resetThreadLogging();
```

Comment:
Are you sure this is wanted? I can't remember how the thread-local logging stuff works, but I would think that we want in some situations at least to be inheriting the logging settings from the thread that is creating me.

Reply:
Despite the name resetThreadLogging() actually set the thread job id to match the current one.  This means it will be done when the thread starts, previously if it wasn't called it would be on first-use.